lib/ui/window.dart

```diff
@@ -269,6 +269,24 @@ class Window {
   double get textScaleFactor => _textScaleFactor;
   double _textScaleFactor = 1.0;
 
+  /// The user setting instructing to always show time using the 24-hour format.
+  /// 
+  /// This option is used by [showTimePicker].
+  bool get alwaysUse24HourFormat => _alwaysUse24HourFormat;
+  bool _alwaysUse24HourFormat = false;
+
+  /// Overrides the value of [alwaysUse24HourFormat] in tests.
+  /// 
+  /// This function may only be called in debug mode. It throws a [StateError]
```

Comment:
I know that Ian isn't terribly fond of debug-only APIs in the dart code.  I tried to do something similar, and he was pretty adamantly against it.

What I proposed doing is adding a testing-only MethodChannel that can be used to have the engine side call back into Dart, simulating real calls from the engine, but only in a test.  Ian was excited about that approach, but I haven't implemented it yet.  That way there don't need to be any public methods added, but we can test messages from the engine.

Reply:
+1

Reply:
Interesting. I've never tried sending something from Dart to C++ over a channel, but I'll give it a shot. One thing I don't know how to implement is "wait for the message to propagate through the engine back to Dart". I guess I can just wait in a looping `Timer`?

Reply:
Channels didn't work. I removed `debugOverrideAlwaysUse24HourFormat`. Will use `MediaQuery` instead.